database/factories/FlightFactory.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Database\Factories;
+
+use Illuminate\Database\Eloquent\Factories\Factory;
+use Lightit\Backoffice\Airlines\Domain\Models\Airline;
+use Lightit\Backoffice\Cities\Domain\Models\City;
+use Lightit\Backoffice\Flights\Domain\Models\Flight;
+
+/**
+ * @extends \Illuminate\Database\Eloquent\Factories\Factory<Flight>
+ */
+class FlightFactory extends Factory
+{
+    protected $model = Flight::class;
+
+    /**
+     * Define the model's default state.
+     *
+     * @return array<string, mixed>
+     */
+    public function definition(): array
+    {
+        $departureTime = fake()->dateTimeBetween('-1 year', '+1 year');
+        $arrivalTime = (clone $departureTime)->modify('+' . fake()->numberBetween(1, 48) . ' hours'); // Add 1 to 12 hours
+
+        /** @var Airline $airline */
+        $airline = AirlineFactory::new()->create();
+
+        /** @var City $originCity */
+        $originCity = CityFactory::new()->create();
+
+        /** @var City $destinationCity */
+        $destinationCity = CityFactory::new()->create();
+
+        return [
+            'airline_id' => $airline->id,
+            'origin_city_id' => $originCity->id,
+            'destination_city_id' => $destinationCity->id,
```

Comment:
```suggestion
        return [
            'airline_id' => AirlineFactory::new(),
            'origin_city_id' => CityFactory::new(),
            'destination_city_id' => CityFactory::new(),
```
In cases like this, where you have an association, assign the instance of the factory in question to the column, and the framework will create the record and assign its ID to the column for you. Not only that, but you prevent your factory from always creating one airline and two cities, even in cases like this, where I want to assign my own created airline and cities to the flight:
```
$airline = AirlineFactory::new()->createOne();

$origin = CityFactory::new()->cresteOne();
$destination = CityFactory::new()->cresteOne();

$flight = FlightFactory::new()->createOne([
    'airline_id' => $airline->id,
    'origin_city_id' => $origin->id,
    'destination_city_id' => $destination->id,
]);
``` 
You'd expect this code to create one airline, two cities, and one flight, but with the current factory implementation, it would create not one but two airlines and four cities.